src/qilib/configuration_helper/instrument_adapter_factory.py

```diff
@@ -32,18 +32,33 @@ class InstrumentAdapterFactory:
     instrument as well as a dictionary that provides key-value pairs of address specifications.
     """
 
-    instrument_adapters: Dict[Tuple[str, str], InstrumentAdapter] = {}
-    _external_adapters: Dict[str, ModuleType] = {}
+    adapter_instances: Dict[Tuple[str, str], InstrumentAdapter] = {}
```

Comment:
The `adapter_instances` is public, but not documented. I prefer docstrings associated to methods or attributes, but it seems there is no concensus on how to do that for class attributes. So I suggest documenting it in the class as
```
class InstrumentAdapterFactory:
"""
   xxx xxx 

   Attributes:
     adapter_instances: xxxx xxx
"""

Reply:
Maybe I should just go through the trouble of making this attribute private. There is no need to keep it public except for some unittests, but I'm sure I can refactor them.